pkg/apis/navigator/validation/elasticsearch_test.go

```diff
@@ -312,52 +332,58 @@ func TestValidateElasticsearchPersistence(t *testing.T) {
 	}
 }
 func TestValidateElasticsearchClusterSpec(t *testing.T) {
-	validPluginsList := []string{"anything"}
-	validNodePool := navigator.ElasticsearchClusterNodePool{
-		Name:         validNodePoolName,
-		Replicas:     validNodePoolReplicas,
-		Roles:        validNodePoolRoles,
-		NodeSelector: validNodePoolNodeSelector,
-		Resources:    validNodePoolResources,
-		Persistence:  validNodePoolPersistenceConfig,
-	}
-	validESImage := navigator.ElasticsearchImage{
-		FsGroup:   int64(1000),
-		ImageSpec: validImageSpec,
-	}
-	validPilotImage := navigator.ElasticsearchPilotImage{
-		ImageSpec: validImageSpec,
-	}
 	errorCases := map[string]navigator.ElasticsearchClusterSpec{
 		"empty spec": {},
 		"valid node pools with duplicate names": {
-			Plugins:   validPluginsList,
-			NodePools: []navigator.ElasticsearchClusterNodePool{validNodePool, validNodePool},
-			Pilot:     validPilotImage,
-			Image:     validESImage,
+			Plugins: validSpecPluginsList,
+			NodePools: []navigator.ElasticsearchClusterNodePool{
+				newValidNodePool("test", 3, navigator.ElasticsearchRoleMaster),
+				newValidNodePool("test", 3, navigator.ElasticsearchRoleMaster),
+			},
+			Pilot:          validSpecPilotImage,
+			Image:          validSpecESImage,
+			MinimumMasters: 4,
 		},
 		"missing pilot image": {
-			Plugins:   validPluginsList,
-			NodePools: []navigator.ElasticsearchClusterNodePool{validNodePool, validNodePool},
-			Image:     validESImage,
+			Plugins:        validSpecPluginsList,
+			NodePools:      []navigator.ElasticsearchClusterNodePool{newValidNodePool("test", 3, navigator.ElasticsearchRoleMaster)},
+			Image:          validSpecESImage,
+			MinimumMasters: 2,
 		},
 		"missing elasticsearch image": {
-			Plugins:   validPluginsList,
-			NodePools: []navigator.ElasticsearchClusterNodePool{validNodePool, validNodePool},
-			Pilot:     validPilotImage,
+			Plugins:        validSpecPluginsList,
+			NodePools:      []navigator.ElasticsearchClusterNodePool{newValidNodePool("test", 3, navigator.ElasticsearchRoleMaster)},
+			Pilot:          validSpecPilotImage,
+			MinimumMasters: 2,
+		},
+		"minimum masters set to low": {
+			Plugins:        validSpecPluginsList,
+			NodePools:      []navigator.ElasticsearchClusterNodePool{newValidNodePool("test", 3, navigator.ElasticsearchRoleMaster)},
+			Pilot:          validSpecPilotImage,
+			Image:          validSpecESImage,
+			MinimumMasters: 1,
+		},
```

Comment:
I wonder if there's ever a case where you want to run with fewer than quorum masters.
I mean if we're allowing a cluster with a single master why not allow a cluster, maybe for testing, with 3 masters but where only one is required to be running?

Reply:
This should never be allowed with Elasticsearch - I've created clusters with 3 nodes and a minimum of 1 master, and immediately got a split brain (two nodes elected themselves leader)